index.js

```diff
@@ -1,18 +1,80 @@
-const readline = require('readline');
+//import the readline library for cleaner output in the terminal
+const readline = require("readline");
 const rl = readline.createInterface(process.stdin, process.stdout);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
     rl.question(questionText, resolve);
   });
 }
+//random number generator
+function randNum(min, max) {
+  let range = max - min + 1;
+  return Math.floor(Math.random() * range) + min;
+}
 
-start();
-
+// have the computer greet the user and ask if they would like to play
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  console.log(
+    "Let's play a game where you (human) make up a number and I (computer) try to guess it. \n\nFirst let's set the range"
+  );
+  // the user sets the range they want to play with
+  let min = 1;
+  let max = await ask("\nWhat would you like the high range to be? "); // I can't think of a user friendly way to ask this!!!
+  let secretNumber = await ask(
+    "\nWhat would you like your number to be?\nI won't peek, I promise...\n"
+  );
+  console.log("You entered: " + secretNumber);
+  secretNumber = parseInt(secretNumber);
+  // the game!
+
+  let number = min + Math.floor((max - min) / 2);
```

Comment:
these variables can be made global for ease of access by other functions; for example, the role reversal story.